src/Common/Date.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Notion\Common;
+
+use DateTimeImmutable;
+
+/**
+ * @psalm-type DateJson = array{ start: string, end?: string }
```

Comment:
Also here, lots of candidates for `@psalm-immutable` :D 


Consider documenting internal symbols that you did not design for end-users as `@psalm-internal Notion\`